### ISSUE TITLE
Automate Unique Resource Group Names with Utility Scripts

### ISSUE THREAD

**Issue by ben-moss** [Azure/sap-hana#348]
## Problem Statement

The `util` scripts provide a simplified interface into using and testing the codebase without over-customization for the user (typically an Engineer). This should allows multiple users to run the same deployments and get the same (or expectedly similar) results. Currently there is an issue when two or more users attempt to deploy into the same subscription, because the templated input JSON files contain hard-coded details - in particular the resource group name, which needs to be unique within an Azure subscription.

## Enhancement

The `util` scripts should be enhanced to support editing the resource group name through a script rather than manual editing (in a similar manner to `util/set_sap_download_credentials.sh`). This could then be either run by an engineer, or embedded as part of the `terraform_v2.sh` wrapper script. Through this, deployments can auto-generate a more-unique resource group name from, for example, properties of the local environment (i.e. environment variables). This follows the existing behaviour of the `util/terraform_v2.sh` script, such that:

When the user runs `util/set_resource_group.sh <resource group name> <target template>`
Then the script:
- [x] Sets the given resource group name in the given template
- [x] Fails with a suitable error when either no resource group name or no target template are provided or the given template does not exist
- [x] When a non-valid set of inputs is provided, the script should provide a list of valid <target template> files based on those available in the templates dir. For example:
  ```text
  Where <JSON template name> is one of the following:
  - clustered_hana
  - rti_only
  - single_node_hana
  ```

## Notes

1. The scripts only have to work for the V2 templates in the standard directory (e.g. single_node_hana, clustered_hana, etc.)
1. To aid future refactoring the code should follow the patterns and conventions already laid out for similar functionality in `terraform_v2.sh`.
1. Some valid sets of inputs are:
   - `util/set_resource_group.sh "rg-dev-hana" single_node_hana`
   - `util/set_resource_group.sh "rg-prd-hana" clustered_hana`
1. Some invalid sets of inputs are:
   - `util/set_resource_group.sh "" single_node_hana`
   - `util/set_resource_group.sh "rg"`
1. There's some existing code within the build pipeline setup that solves the same kind of problem to allow multiple pipelines to deploy in parallel (these solution could be consolidated)

## Dependencies

None, but relates to #340 Capability to specify target input JSON for set_sap_download_credentials.sh

## Future Work

1. Generalize the ability to programatically edit the JSON templates for other use cases, such as:
   1. Switching between SLES and RHEL
   1. Setting HANA version
   1. Setting the SID/Instance Number
   1. Switching the ansible execution on/off
1. Integrate the use of these scripts within the pipeline to create more readable pipeline test scenarios

## Checklist

- [x] Usage documentation updated as necessary
- [x] Architecture documentation updated as necessary

## References

None
